lisa/tests/base.py

```diff
@@ -905,10 +931,33 @@ def rtapp_profile(self):
     @property
     def rtapp_tasks(self):
         """
-        Sorted list of rtapp task names, as defined in ``rtapp_profile``
-        attribute.
+        The rtapp task names as found from the trace in this bundle.
+
+        :return: the list of actual trace task names
         """
-        return sorted(self.rtapp_profile.keys())
+        task_map = getattr(self, 'task_map', None)
+        if not task_map:
+            raise RuntimeError("Task name mapping not available: use (get_)trace() before.")
+
+        return sorted(itertools.chain.from_iterable(self.task_map.values()))
+
+    def rtapp_tasks_of(self, profile_name):
```

Comment:
None of the errors handling of that method are needed, just exposing the dict(profile_keys, list(comm)) we compute is enough. Client code can rely on the fact that the keys are the same as in the profile since we check it, so it's already straightforward to use.